sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -56,18 +58,36 @@ process.on("exit", (code: number) => {
 
 const providerCache: { [key: string]: dynamic.ResourceProvider } = {};
 
-function getProvider(props: any): dynamic.ResourceProvider {
+// getProvider deserializes the provider from the string found in
+// `props[providerKey]` and calls `provider.configure` with the config. The
+// deserialized and configured provider is stored in `providerCache`. This
+// guarantees that the provider is only deserialized and configured once per
+// process.
+function getProvider(props: any, rawConfig: Record<string, any>): dynamic.ResourceProvider {
     const providerString = props[providerKey];
     let provider: any = providerCache[providerString];
     if (!provider) {
         provider = requireFromString(providerString).handler();
         providerCache[providerString] = provider;
+        if (provider.configure) {
+            const config = new dynConfig.Config(rawConfig, getProject());
+            const req = new ConfigureRequest(config);
+            provider.configure(req);
+        }
     }
 
     // TODO[pulumi/pulumi#414]: investigate replacing requireFromString with eval
     return provider;
 }
 
+class ConfigureRequest implements dynamic.ConfigureRequest {
+    public readonly config: dynamic.Config;
+
+    constructor(config: dynamic.Config) {
+        this.config = config;
+    }
+}
```

Comment:
Does this class buy us anything over just instantiating an object and asserting it has the interface type? E.g. `const req: dynamic.ConfigureRequest = { config }`.

Reply:
Nope. I'll get rid of it!